core/src/test/java/org/elasticsearch/action/bulk/BulkRequestTests.java

```diff
@@ -255,4 +255,24 @@ public void testSmileIsSupported() throws IOException {
         assertEquals(1, request.sourceAsMap().size());
         assertEquals("value", request.sourceAsMap().get("field"));
     }
+
+    public void testToValidateUpsertRequestAndVersionInBulkRequest() {
+        UpdateRequest updateRequest = new UpdateRequest("index", "type", "id");
+        updateRequest.version(1L);
+        updateRequest.doc("{}", XContentType.JSON);
+        updateRequest.upsert(new IndexRequest("index","type", "id").source("{}", XContentType.JSON));
+        BulkRequest bulkRequest = new BulkRequest();
```

Comment:
I meant a test that uses the bulk request's overrides in `BulkRequest#add`. That way we are testing a way of "sneaking" the version into the request that is unique to bulk requests.